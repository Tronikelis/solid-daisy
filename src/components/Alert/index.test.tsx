import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { ComponentProps } from "solid-js";
import { beforeEach, describe, expect, it } from "vitest";

import { Alert } from ".";

describe("<Alert />", () => {
    let props: ComponentProps<typeof Alert>;

    beforeEach(() => {
        props = {
            children: "foo",
        };
    });

    it("renders correctly", () => {
        const { container } = render(() => <Alert {...props} />);
        expect(container).toMatchSnapshot();
    });

    it("renders icon", () => {
        const screen = render(() => <Alert {...props} icon={<div>icon</div>} />);

        expect(screen.getByText("icon")).toBeInTheDocument();
    });

    it("closes itself after pressing x", async () => {
        const screen = render(() => <Alert {...props} />);

        expect(screen.getByText("foo")).toBeInTheDocument();

        await userEvent.click(screen.getByText("âœ•"));

        expect(screen.queryByText("foo")).not.toBeInTheDocument();
    });
});
