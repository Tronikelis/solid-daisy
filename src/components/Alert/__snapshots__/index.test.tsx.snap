// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Alert /> > renders correctly 1`] = `
<div>
  <div
    class="alert"
  >
    
    foo
    <div>
      <button
        class="btn btn-sm btn-circle btn-ghost"
        type="button"
      >
        
        
        âœ•
        
      </button>
    </div>
  </div>
</div>
`;
